### DISCUSSION TITLE
👋 help wanted with local DNS resolve

### DISCUSSION THREAD

**Discussion by dark-age** [0xERR0R/blocky#1252] [General]
Hello,
I've migrated from Pi-hole and want to set up Blocky as my primary local DNS resolver.

Windows returns the following and I'm not able to visit my nas via the custom domain name

C:\Windows\System32>nslookup nas.local.lan
Server:  UnKnown
Address:  192.168.86.2

Non-authoritative answer:
Name:    nas.local.lan
Addresses:  ::17
          0.0.0.17

Blocky Config

customDNS:
  customTTL: 1h
  filterUnmappedTypes: true
  mapping:
    nas.local.lan: 192.168.86.12

**Comment by 0xERR0R** [comment]
I tried v0.21 with following config:

```yaml
upstream:
  default:
    - 5.9.164.112
    - 1.1.1.1
customDNS:
  mapping:
    nas.local.lan: 192.168.86.12
port: 55555
```

I can resolve this custom address with dig on linux

```sh
dig @localhost -p55555 nas.local.lan

; <<>> DiG 9.18.17 <<>> @localhost -p55555 nas.local.lan
; (2 servers found)
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 51407
;; flags: qr rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 0

;; QUESTION SECTION:
;nas.local.lan.                 IN      A

;; ANSWER SECTION:
nas.local.lan.          3600    IN      A       192.168.86.12

;; Query time: 0 msec
;; SERVER: ::1#55555(localhost) (UDP)
;; WHEN: Tue Aug 15 22:59:07 CEST 2023
;; MSG SIZE  rcvd: 47

```
